atlasdb-timelock-server/src/integTest/java/com/palantir/atlasdb/timelock/PaxosTimeLockServerIntegrationTest.java

```diff
@@ -0,0 +1,170 @@
+        FileUtils.deleteDirectory(new File("var/test/data"));
+    }
+
+    @Test
+    public void lockServiceShouldAllowUsToTakeOutLocks() throws InterruptedException {
+        RemoteLockService lockService = getLockService(CLIENT_1);
+
+        LockRefreshToken token = lockService.lock(LOCK_CLIENT_NAME, LockRequest.builder(LOCK_MAP)
+                .doNotBlock()
+                .build());
+
+        assertThat(token).isNotNull();
+
+        lockService.unlock(token);
+    }
+
+    @Test
+    public void lockServiceShouldAllowUsToTakeOutSameLockInDifferentNamespaces() throws InterruptedException {
+        RemoteLockService lockService1 = getLockService(CLIENT_1);
+        RemoteLockService lockService2 = getLockService(CLIENT_2);
+
+        LockRefreshToken token1 = lockService1.lock(LOCK_CLIENT_NAME, LockRequest.builder(LOCK_MAP)
+                .doNotBlock()
+                .build());
+        LockRefreshToken token2 = lockService2.lock(LOCK_CLIENT_NAME, LockRequest.builder(LOCK_MAP)
+                .doNotBlock()
+                .build());
+
+        assertThat(token1).isNotNull();
+        assertThat(token2).isNotNull();
+
+        lockService1.unlock(token1);
+        lockService2.unlock(token2);
+    }
+
+    @Test
+    public void lockServiceShouldNotAllowUsToRefreshLocksFromDifferentNamespaces() throws InterruptedException {
+        RemoteLockService lockService1 = getLockService(CLIENT_1);
+        RemoteLockService lockService2 = getLockService(CLIENT_2);
+
+        LockRefreshToken token1 = lockService1.lock(LOCK_CLIENT_NAME, LockRequest.builder(LOCK_MAP)
+                .doNotBlock()
+                .build());
+
+        assertThat(token1).isNotNull();
+        assertThat(lockService1.refreshLockRefreshTokens(ImmutableList.of(token1))).isNotEmpty();
+        assertThat(lockService2.refreshLockRefreshTokens(ImmutableList.of(token1))).isEmpty();
+
+        lockService1.unlock(token1);
+    }
+
+    @Test
+    public void timestampServiceShouldGiveUsIncrementalTimestamps() {
+        TimestampService timestampService = getTimestampService(CLIENT_1);
+
+        long timestamp1 = timestampService.getFreshTimestamp();
+        long timestamp2 = timestampService.getFreshTimestamp();
+
+        assertThat(timestamp1).isLessThan(timestamp2);
+    }
+
+    @Test
+    public void timestampServiceShouldRespectDistinctClientsWhenIssuingTimestamps() {
+        TimestampService timestampService1 = getTimestampService(CLIENT_1);
+        TimestampService timestampService2 = getTimestampService(CLIENT_2);
+
+        long firstServiceFirstTimestamp = timestampService1.getFreshTimestamp();
+        long secondServiceFirstTimestamp = timestampService2.getFreshTimestamp();
+
+        long firstServiceSecondTimestamp = timestampService1.getFreshTimestamp();
+        long secondServiceSecondTimestamp = timestampService2.getFreshTimestamp();
+
+        assertThat(firstServiceFirstTimestamp + 1).isEqualTo(firstServiceSecondTimestamp);
```

Comment:
the only guarantee that we need is that the timestamps are strictly higher. in practice for the current implementation, without nodes dying, this is the case.

we are testing that timestamps rise independently for different clients, but i think that's a difficult test if we don't also leverage the fact that the timestamps are incrementing by one.